app/controllers/projects_controller.rb

```diff
@@ -37,6 +37,14 @@ def show
     @project_json = @project.to_json(
       methods: %i[histories memberships metadata components available_components available_members]
     )
+    project = JSON.parse(@project_json)
+    project[:details] = {
+      locked: @project.rules.where(locked: true).size,
+      review: @project.rules.where(changes_requested: true).size,
+      completed: @project.rules.where(locked: true).size,
+      total: @project.rules.size
+    }
+    @project_json = project.to_json
```

Comment:
Are line 40 and 47 really necessary? it looks like `@project_json` is already set.

Reply:
that line is necessary, but I rewrote it in a less confusing way: https://github.com/mitre/vulcan/pull/343/commits/f8f5edef0704efd55becb956786b1a9141fcb283